plugins/mongodb/mongodbembed.hpp

```diff
@@ -0,0 +1,736 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _MONGODBEMBED_INCL
+#define _MONGODBEMBED_INCL
+
+#ifdef MONGODBEMBED_PLUGIN_EXPORTS
+#define MONGODBEMBED_PLUGIN_API DECL_EXPORT
+#else
+#define MONGODBEMBED_PLUGIN_API DECL_IMPORT
+#endif
+
+// Using cpp driver from http://mongocxx.org/mongocxx-v3/installation/linux/
+#include <cstdint>
+#include "mongocxx/client.hpp"
+#include "mongocxx/instance.hpp"
+#include "mongocxx/logger.hpp"
+#include "mongocxx/pool.hpp"
+#include "mongocxx/uri.hpp"
+#include "mongocxx/options/client.hpp"
+#include "mongocxx/stdx.hpp"
+#include "mongocxx/cursor.hpp"
+#include "mongocxx/exception/bulk_write_exception.hpp"
+#include "mongocxx/exception/error_code.hpp"
+#include "mongocxx/exception/logic_error.hpp"
+#include "mongocxx/exception/operation_exception.hpp"
+#include "mongocxx/exception/server_error_code.hpp"
+#include "bsoncxx/json.hpp"
+#include "bsoncxx/builder/stream/helpers.hpp"
+#include "bsoncxx/builder/stream/array_context.hpp"
+#include "bsoncxx/builder/stream/document.hpp"
+#include "bsoncxx/builder/stream/array.hpp"
+#include "bsoncxx/document/value.hpp"
+#include "bsoncxx/document/view.hpp"
+#include "bsoncxx/stdx/make_unique.hpp"
+#include "bsoncxx/stdx/optional.hpp"
+#include "bsoncxx/stdx/string_view.hpp"
+#include "bsoncxx/builder/basic/array.hpp"
+#include "bsoncxx/builder/basic/document.hpp"
+#include "bsoncxx/builder/basic/kvp.hpp"
+#include "bsoncxx/types.hpp"
+
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <memory>
+
+#include "platform.h"
+#include "jthread.hpp"
+#include "hqlplugins.hpp"
+#include "eclrtl_imp.hpp"
+#include "eclhelper.hpp"
+#include "rtlembed.hpp"
+#include "jptree.hpp"
+#include "tokenserialization.hpp"
+#include "rtlds_imp.hpp"
+#include "rtlfield.hpp"
+#include "roxiemem.hpp"
+#include <time.h>
+
+#include <vector>
```

Comment:
Comment:  We normally try to minimize the included headers by shifting everything possible to the .cpp file.  If you have time, it may be worthwhile to go through that exercise.